servicetalk-loadbalancer-experimental-provider/src/main/java/io/servicetalk/loadbalancer/experimental/DefaultHttpLoadBalancerProvider.java

```diff
@@ -0,0 +1,119 @@
+import io.servicetalk.transport.api.HostAndPort;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * A client builder provider that supports enabling the new `DefaultLoadBalancer` in applications via property flags.
+ * See the packages README.md for more details.
+ */
+public class DefaultHttpLoadBalancerProvider implements HttpProviders.SingleAddressHttpClientBuilderProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultHttpLoadBalancerProvider.class);
+
+    private final DefaultLoadBalancerProviderConfig config;
+
+    public DefaultHttpLoadBalancerProvider() {
+        this(DefaultLoadBalancerProviderConfig.INSTANCE);
+    }
+
+    // exposed for testing
+    DefaultHttpLoadBalancerProvider(final DefaultLoadBalancerProviderConfig config) {
+        this.config = requireNonNull(config, "config");
+    }
+
+    @Override
+    public final <U, R> SingleAddressHttpClientBuilder<U, R> newBuilder(U address,
+                                                                  SingleAddressHttpClientBuilder<U, R> builder) {
+        final String serviceName = clientNameFromAddress(address);
+        if (config.enabledForServiceName(serviceName)) {
+            try {
+                HttpLoadBalancerFactory<R> loadBalancerFactory = DefaultHttpLoadBalancerFactory.Builder.<R>from(
+                        defaultLoadBalancer(serviceName)).build();
+                builder = builder.loadBalancerFactory(loadBalancerFactory);
+                return new LoadBalancerIgnoringBuilder(builder, serviceName);
+            } catch (Throwable ex) {
+                LOGGER.warn("Failed to enabled DefaultLoadBalancer for client to address {}.", address, ex);
+            }
+        }
+        return builder;
+    }
+
+    private <R> LoadBalancerFactory<R, FilterableStreamingHttpLoadBalancedConnection> defaultLoadBalancer(
+            String serviceName) {
+        return LoadBalancers.<R, FilterableStreamingHttpLoadBalancedConnection>
+                        builder("experimental-load-balancer")
+                .loadBalancerObserver(new DefaultLoadBalancerObserver(serviceName))
+                // set up the new features.
+                .outlierDetectorConfig(config.outlierDetectorConfig())
+                .loadBalancingPolicy(config.getLoadBalancingPolicy())
+                .build();
+    }
+
+    /**
+     * Extract the service name from the address object.
+     * Note: this is a protected method to allow overriding for custom address types.
+     * @param <U> the unresolved type of the address.
+     * @param address the address from which to extract the service name.
+     * @return the String representation of the provided address.
+     */
+    protected <U> String clientNameFromAddress(U address) {
+        String serviceName;
+        if (address instanceof HostAndPort) {
+            serviceName = ((HostAndPort) address).hostName();
```

Comment:
the code matches the documentation above, would it be useful in a HostAndPort case to allow the user to specify `hostName:port` as the serviceName to distinguish different ports?

Reply:
I don't know that it will be that useful in practice and this matches other behavior I've seen.

Reply:
I assume these are the only two types of addresses other providers support.

Reply:
👍